Tests/Shared.Specs/BasicEquivalencySpecs.cs

```diff
@@ -3334,6 +3335,40 @@ public void
             act.Should().NotThrow();
         }
 
+        [Fact]
+        public void When_an_enumerable_collection_returns_itself_it_should_detect_the_cyclic_reference()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            var instance1 = new SelfReturningEnumerable();
+            var instance2 = new SelfReturningEnumerable();
+            var actual = new HashSet<SelfReturningEnumerable> {instance1, instance2};
```

Comment:
❓ Does this need to be a `HashSet`, or is that just because the example in the issue used it?
Enumeration of a `HashSet` does not guarantee a particular order.

Reply:
Copied the example from the original issue, but a `List<>` does work just as well.